File: core/src/main/java/fr/erictruong/android/core/adapter/SimpleFragmentPagerAdapter.java
package fr.erictruong.android.core.adapter;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import java.util.ArrayList;

public class SimpleFragmentPagerAdapter extends FragmentPagerAdapter {

    private ArrayList<String> titles;
    private ArrayList<Fragment> fragments;

    public SimpleFragmentPagerAdapter(FragmentManager fm) {
        super(fm);
        titles = new ArrayList<>(3);
        fragments = new ArrayList<>(3);
    }

    public void addFragment(String title, Fragment fragment) {
        titles.add(title);
        fragments.add(fragment);
    }

    public void addFragment(Fragment fragment) {
        titles.add(null);
        fragments.add(fragment);
    }

    @Override
    public Fragment getItem(int position) {
        return fragments.get(position);
    }

    @Override
    public int getCount() {
        return titles.size();
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return titles.get(position);
    }
}

File: core/src/main/java/fr/erictruong/android/core/adapter/SimpleFragmentPagerAdapter.kt
package fr.erictruong.android.core.adapter

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter

import java.util.ArrayList

class SimpleFragmentPagerAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {

    private val titles: ArrayList<String>
    private val fragments: ArrayList<Fragment>

    init {
        titles = ArrayList<String>()
        fragments = ArrayList<Fragment>()
    }

    fun addFragment(title: String, fragment: Fragment) {
        titles.add(title)
        fragments.add(fragment)
    }

    fun addFragment(fragment: Fragment) {
        titles.add("")
        fragments.add(fragment)
    }

    override fun getItem(position: Int): Fragment {
        return fragments[position]
    }

    override fun getCount(): Int {
        return titles.size
    }

    override fun getPageTitle(position: Int): CharSequence {
        return titles[position]
    }
}
